### PR TITLE
add xorMonoid

### PR DESCRIPTION
**Pull request by drostron**

### CODE REVIEW

File: tests/shared/src/test/scala/cats/tests/XorTests.scala [hunk 1]
```diff
@@ -12,6 +13,8 @@ import org.scalacheck.Arbitrary._
 import scala.util.Try
 
 class XorTests extends CatsSuite {
+  checkAll("Xor[String, Int]", algebra.laws.GroupLaws[Xor[String, Int]].monoid)
```
Comment (adelbertc) [inline tests/shared/src/test/scala/cats/tests/XorTests.scala]: Check Serializability

Comment (drostron) [inline tests/shared/src/test/scala/cats/tests/XorTests.scala]: I see `- Xor[String, Int].monoid.serializable` in the test output. It appears serializability is checked via:

https://github.com/non/algebra/blob/v0.3.1/laws/shared/src/main/scala/algebra/laws/GroupLaws.scala#L28

Is that sufficient or is there a reason to have the following in addition?

```scala
checkAll("Xor[String, Int]", SerializableTests.serializable(Monoid[Xor[String, Int]]))
```

Comment (adelbertc) [inline tests/shared/src/test/scala/cats/tests/XorTests.scala]: Oh you're right, `algebra` includes `Serializable` checks in their laws - @non confirm?

**Comment by adelbertc** [comment]
:+1:

**Comment by non** [comment]
:+1:

**Pull request merged**